### PR TITLE
truncate overlong changelogs to 100 lines + last line

### PR DESCRIPTION
**Pull request by cboltz**
This fixes the multi-megabyte "New Tumbleweed snapshot" mails if the changelog comparison can't find a common changelog entry to stop at, or if a package simply has a really big changelog since the last submit request.

WARNING: I don't have test data to test my changes, therefore I wrote the truncation code more or less blindfold. In theory it should work, but in practise I'd seriously recommend to test it before merging this pull request.

### CODE REVIEW

**Comment by lnussel** [comment]
i2 already contains the line count

**Comment by cboltz** [comment]
> should add a break if i2 > 100

Well, depends - an interesting detail in my code is that it will create something like that:

```
line 0
line 1...99
line 100
  ... changelog too long, skipping 42 lines ...
last line
```

(`line 1...99` summarized for this comment - in realitiy, there would be 99 lines of changelog)

If using a `break`, it would IMHO (please correct me if I'm wrong) be impossible to know how many lines were skipped, and what the last line was. Both are useful to know how much of the changelog was skipped.

File: factory-package-news/factory-package-news.py [hunk 1]
```diff
@@ -239,10 +241,21 @@ def do_diff(self, subcmd, opts, version1, version2):
...
+
+            changedetails = ""
             for (i2, t2) in enumerate(v2changelogs[srpm]['changelogtime']):
                 if t2 == t1:
                     break
-                details += "\n" + v2changelogs[srpm]['changelogtext'][i2]
+                changedetails += "\n" + v2changelogs[srpm]['changelogtext'][i2]
+
+            # if a changelog is too long, cut it off after changelog_max_lines lines
+            changedetails_lines = changedetails.split('\n')
```
[the remark this answers is not in the document]
Comment (cboltz) [reply factory-package-news/factory-package-news.py]: Good idea, done.

**Comment by cboltz** [comment]
@lnussel please read my response to your review request ;-)
(and sorry that it took so long to respond!)

**Comment by coolo** [comment]
@DimStar77 Before we go into another 2 year for this PR, could you chime in?

**Comment by DimStar77** [comment]
> @DimStar77 Before we go into another 2 year for this PR, could you chime in?

Generally, I am much in favor of this; some packages submit updates only after long integration phases (think e.g. GNOME 41 -> 42) where the changelog accumulates a lot of info; not all of that is relevant in the announcement mails.

So shortening them to 100 lines is reasonable

**Comment by coolo** [comment]
Let's get over this

**Pull request merged**